lib/auth/bot.go

```diff
@@ -216,6 +234,28 @@ func (a *Server) generateInitialBotCerts(ctx context.Context, botName, username,
 	var generation uint64
 	if renewable {
 		generation = 1
+		initialAuth.Generation = 1
+	}
+
+	var botInstanceID string
+	if experiment.Enabled() {
+		uuid, err := uuid.NewRandom()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		bi := newBotInstance(&machineidv1.BotInstanceSpec{
+			BotName:    botName,
+			InstanceId: uuid.String(),
+
+			// TODO: set TTL? should we rely on the known expiration instead?
+			// (auth may overwrite the value now or later)
```

Comment:
Not 100% sure, but I may regret introducing a TTL field to the bot spec given every renewal provides an expiry. We'll need to pass that along somehow, or come up with a sensible value for this field another way.

Reply:
It feels to me like we could just directly leverage the metadata.expiry field, I'm not too sure having a separate TTL field has any positive impact - no harm in dropping it from the protos in one of these PRs.

Reply:
Yeah, I'll go ahead and make that change. I figure we're safe to make breaking changes to the protos still, since they should never have been exercised on any real clusters.